sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -65,21 +83,21 @@ def process_content(data, start_offset, end_offset, encryption):
 class _ChunkDownloader(object):  # pylint: disable=too-many-instance-attributes
     def __init__(
         self,
-        client=None,
-        non_empty_ranges=None,
-        total_size=None,
-        chunk_size=None,
-        current_progress=None,
-        start_range=None,
-        end_range=None,
-        stream=None,
-        parallel=None,
-        validate_content=None,
-        encryption_options=None,
-        encryption_data=None,
-        progress_hook=None,
-        **kwargs
-    ):
+        client: "BlobOperations" = None,  # type: ignore [assignment]
+        non_empty_ranges: Optional[List[Dict[str, Any]]] = None,
+        total_size: int = None,  # type: ignore [assignment]
+        chunk_size: int = None,  # type: ignore [assignment]
+        current_progress: int = None,  # type: ignore [assignment]
+        start_range: int = None,  # type: ignore [assignment]
+        end_range: int = None,  # type: ignore [assignment]
+        validate_content: bool = None,  # type: ignore [assignment]
+        stream = None,
+        parallel: Optional[int] = None,
+        encryption_options: Dict[str, Any] = None,  # type: ignore [assignment]
```

Comment:
Didn't we discuss a way to get rid of all these ignores? This is a private class so you can change this constructor if needed to not have optional params.